Modules/_decimal/_decimal.c

```diff
@@ -548,25 +548,33 @@ getround(PyObject *v)
    initialized to new SignalDicts. Once a SignalDict is tied to
    a context, it cannot be deleted. */
 
-static mpd_context_t dflt_ctx;
-
 static int
 signaldict_init(PyObject *self, PyObject *args UNUSED, PyObject *kwds UNUSED)
 {
-    SdFlagAddr(self) = &dflt_ctx.status;
+    SdFlagAddr(self) = NULL;
     return 0;
 }
 
 static Py_ssize_t
-signaldict_len(PyObject *self UNUSED)
+signaldict_len(PyObject *self)
 {
+    if (SdFlagAddr(self) == NULL) {
+        PyErr_SetString(PyExc_ValueError,
```

Comment:
how about using this function for consistency?
https://github.com/python/cpython/blob/773167e8d2f4fedfaaf9570019d7829ae3e37faf/Modules/_decimal/_decimal.c#L250-L255

Reply:
Great. How about adding a function named `value_error ptr` and macro `INVALID_SIGNALDICT_PTR`? 
This would reduce duplication, similar to the function `runtime_error_ptr` and macro `INTERNAL_ERROR_PTR`.

Reply:
Great!